compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -308,11 +308,7 @@ types_h export_attribute userHeaders enumHeader allocator alloc_ctors_enabled ty
             -- even if implicit would be okay, fall back to explicit for
```

Comment:
The [`struct _bond_vc12_ctor_workaround_`](https://github.com/microsoft/bond/blob/65017969ce65322a60d8314b0bf65c5e99ed8ee9/compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs#L219) above can also be removed ([cannot comment on collapsed lines](https://github.com/isaacs/github/issues/1655) 😞).

Reply:
@ara-ayvazyan - Yeah, I was going to do another round of cleanup after this one. Would you prefer to have it done in this PR?

Reply:
I thought that part seem to be related to this PR.